app/announcement/models/Need.py

```diff
@@ -2,33 +2,45 @@
 Module defining the Need model.
 
 This module contains the Need class, which represents a specific need or requirement
-in the application, inheriting from the LabeledIcon base class.
+in the application base class.
 """
 
 from django.db import models
 
-from app.core.models import LabeledIcon
 from app.core.models.ArchitectSpeciality import ArchitectSpeciality
 
 
-class Need(LabeledIcon):
+class Need(models.Model):
     """
-    Model representing a specific need or requirement, inheriting from LabeledIcon.
+    Model representing a specific need or requirement.
+
 
-    Inherits:
-        LabeledIcon: Base class providing fields for label and icon.
 
     Attributes:
         architect_speciality (ForeignKey): Specialization of the architect related to this need.
+        label
+        icon
     """
 
+    label = models.CharField(max_length=255, default="")
+    icon = models.ImageField(upload_to="NeedIcons/")
```

Comment:
You should specify a folder for this type of icon and then split it depending on the usage

Reply:
Can you explain more please, the icon images are uploaded to media/NeedIcons , where should they be uploaded?